apps/prairielearn/assets/scripts/instructorAssessmentQuestionsClient.ts

```diff
@@ -12,13 +82,562 @@ onDocumentReady(() => {
+          ? zones[zoneNumber].questions[questionNumber].alternatives[alternativeNumber]
+          : zones[zoneNumber].questions[questionNumber],
+        urlPrefix,
+        assessmentInstanceId,
+        zoneIndex: zoneNumber,
+        questionIndex: questionNumber,
+        alternativeIndex: alternativeNumber,
+        assessmentType,
+      }).toString()),
+    );
+    $('#editQuestionModal').modal('show');
+    enableGradingOptions(
+      (document.getElementById('gradingMethod') as HTMLSelectElement)?.value ?? 'auto',
+    );
+    $('#editQuestionModal').modal('show');
+  });
+
+  on('input', '#qidInput', (e) => {
+    const qidInput = e.target as HTMLInputElement;
+    const updateQuestionButton = document.getElementById('updateQuestionButton');
+    if (qidInput.value) {
+      updateQuestionButton?.removeAttribute('disabled');
+    } else {
+      updateQuestionButton?.setAttribute('disabled', 'true');
+    }
+  });
+
+  on('click', '.add-question', (e) => {
+    const addButton = (e.target as HTMLElement).closest('button');
+    const zoneIndex = addButton?.dataset.zoneIndex ?? '0';
+    const questionIndex = addButton?.dataset.questionIndex ?? '0';
+    const alternativeIndex = (e.target as HTMLElement).dataset.alternativeIndex ?? '0';
+    $('#editQuestionModal').replaceWith(
+      (document.createElement('div').innerHTML = EditQuestionModal({
+        newQuestion: true,
+        urlPrefix,
+        assessmentInstanceId,
+        zoneIndex: parseInt(zoneIndex),
+        questionIndex: parseInt(questionIndex),
+        alternativeIndex: parseInt(alternativeIndex),
+        assessmentType,
+      }).toString()),
+    );
+    $('#editQuestionModal').modal('show');
+    enableGradingOptions('auto');
+  });
+
+  on('click', '#findQid', async () => {
+    $('#editQuestionModal').modal('hide');
+    const response = await fetch(
+      `${urlPrefix}/assessment/${assessmentInstanceId}/questions/findqid`,
+    );
+    const findQidHtml = await response.text();
+    $('#findQIDModal').replaceWith(
+      (document.createElement('div').innerHTML = findQidHtml.toString()),
+    );
+    $('#findQIDModal').modal('show');
+  });
+
+  function enableGradingOptions(method: string) {
+    if (method === 'auto') {
+      document.querySelectorAll('.hw-auto-points').forEach((el) => {
+        el.removeAttribute('hidden');
+      });
+      document.querySelectorAll('.hw-manual-points').forEach((el) => {
+        el.setAttribute('hidden', 'true');
+      });
+    } else if (method === 'manual') {
+      document.querySelectorAll('.hw-auto-points').forEach((el) => {
+        el.setAttribute('hidden', 'true');
+      });
+      document.querySelectorAll('.hw-manual-points').forEach((el) => {
+        el.removeAttribute('hidden');
+      });
+    }
+  }
+
+  on('change', '#gradingMethod', (e) => {
+    const method = (e.target as HTMLSelectElement).value;
+    enableGradingOptions(method);
+  });
+
+  on('click', '#updateQuestionButton', async (e) => {
```

Comment:
Self review: I believe this function has redundancies and can be reworked to be much cleaner, it is just currently functional. I will work on making this cleaner and easier to read early next week.